File: python/problem02.py
#!/usr/bin/env python

elem = 0
seq = [1, 2]

while(elem < 4000000):
    elem = seq[-1] + seq[-2]
    seq.append(elem)

even = [x for x in seq if x % 2 == 0]

print even
print sum(even)

File: elixir/problem02.exs
#!/usr/bin/env elixir

seq = [1,2,3]

defmodule Fib do
  def fib([h|t]) do
    elem = Enum.at(t, -1) + Enum.at(t, -2)
    if elem < 4000000 do
      fib([h] ++ t ++ [elem])
    else
      [h] ++ t
    end
  end
end

list = Fib.fib(seq)
even = lc n inlist list, rem(n, 2) == 0, do: n
sum = Enum.reduce(even, 0, fn(x, acc) -> x + acc end)

IO.puts "Sum: #{sum}"
